### DISCUSSION TITLE
HTTP Endpoint Issue {System.Dynamic.ExpandoObject}

### DISCUSSION THREAD

**Discussion by vabs212** [elsa-workflows/elsa-core#1422] [Design & tech]
I have a problem in elsa. I have made custom activity in Elsa .And now creating a workflow in Dashboard UI.There i have made a HTTP endpoint Dashboard which on invoked will then go to the Actviity 1. However when I  go in my activity i am not able to get the body of the API called > ia m getting some dynamic object in which data i stored in KEY and value pair .but i  want the data in  my custom model format. I have mentioned the target type also while specifying the API still not getting the data in required format.  i am getting data in {System.Dynamic.ExpandoObject}
Please help .here is my workflow json.
{
  "$id": "1",
  "definitionId": "7724e9a9209b4434b6776e101c366b18",
  "versionId": "c6146984dbfb4ea98823fbcfb4dbf5f5",
  "name": "UserRegistrationWorkFlow",
  "displayName": "UserRegistrationWorkFlow",
  "description": "UserRegistrationWorkFlow",
  "version": 2,
  "variables": {
    "$id": "2",
    "data": {}
  },
  "customAttributes": {
    "$id": "3",
    "data": {}
  },
  "contextOptions": {
    "$id": "4",
    "contextFidelity": "Burst"
  },
  "isSingleton": false,
  "persistenceBehavior": "Suspended",
  "deleteCompletedInstances": false,
  "isPublished": true,
  "isLatest": true,
  "activities": [
    {
      "$id": "5",
      "activityId": "1c871c36-64e3-4890-9b3d-5a749c0d6d8e",
      "type": "HttpEndpoint",
      "name": "HTTP Endpoint",
      "displayName": "HTTP Endpoint",
      "persistWorkflow": true,
      "loadWorkflowContext": true,
      "saveWorkflowContext": true,
      "properties": [
        {
          "$id": "6",
          "name": "Path",
          "expressions": {
            "$id": "7",
            "Literal": "/test"
          }
        },
        {
          "$id": "8",
          "name": "Methods",
          "expressions": {
            "$id": "9",
            "Json": "[\"POST\"]"
          }
        },
        {
          "$id": "10",
          "name": "ReadContent",
          "expressions": {
            "$id": "11",
            "Literal": "true"
          }
        },
        {
          "$id": "12",
          "name": "TargetType",
          "expressions": {
            "$id": "13",
            "Literal": "RegistrationModel"
          }
        }
      ],
      "propertyStorageProviders": {}
    },
    {
      "$id": "14",
      "activityId": "191f0857-9f14-46b3-aaef-cb5f2d2a1735",
      "type": "CreateUser",
      "name": "Create User",
      "displayName": "Create User",
      "description": "Create User",
      "persistWorkflow": false,
      "loadWorkflowContext": true,
      "saveWorkflowContext": false,
      "properties": [],
      "propertyStorageProviders": {}
    },
    {
      "$id": "15",
      "activityId": "1e574748-7698-4106-b0ed-26cb5eefe402",
      "type": "Fork",
      "displayName": "Fork",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "properties": [
        {
          "$id": "16",
          "name": "Branches",
          "expressions": {
            "$id": "17",
            "Json": "[\"ValidateUsingEmail\",\"ValidateUsingOtp\"]"
          }
        }
      ],
      "propertyStorageProviders": {}
    },
    {
      "$id": "18",
      "activityId": "949fe4bf-1261-440b-a836-137a4f7ebd2a",
      "type": "ActivateUserEmail",
      "name": "Activate User Email",
      "displayName": "Activate User Email",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "properties": [],
      "propertyStorageProviders": {}
    },
    {
      "$id": "19",
      "activityId": "e5656747-01e2-4a6f-a9f5-f36fd9a2bfed",
      "type": "ActivateUserSms",
      "name": "Activate User Sms",
      "displayName": "Activate User Sms",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "properties": [],
      "propertyStorageProviders": {}
    }
  ],
  "connections": [
    {
      "$id": "20",
      "sourceActivityId": "1c871c36-64e3-4890-9b3d-5a749c0d6d8e",
      "targetActivityId": "191f0857-9f14-46b3-aaef-cb5f2d2a1735",
      "outcome": "Done"
    },
    {
      "$id": "21",
      "sourceActivityId": "191f0857-9f14-46b3-aaef-cb5f2d2a1735",
      "targetActivityId": "1e574748-7698-4106-b0ed-26cb5eefe402",
      "outcome": "Done"
    },
    {
      "$id": "22",
      "sourceActivityId": "1e574748-7698-4106-b0ed-26cb5eefe402",
      "targetActivityId": "949fe4bf-1261-440b-a836-137a4f7ebd2a",
      "outcome": "ValidateUsingEmail"
    },
    {
      "$id": "23",
      "sourceActivityId": "1e574748-7698-4106-b0ed-26cb5eefe402",
      "targetActivityId": "e5656747-01e2-4a6f-a9f5-f36fd9a2bfed",
      "outcome": "ValidateUsingOtp"
    }
  ],
  "id": "c6146984dbfb4ea98823fbcfb4dbf5f5"
}

**Comment by sfmskywalker** [comment]
When you specify the **Target Type** value. make sure you provide the assembly-qualified type name. For example: `MyApp.Models.RegistrationModel, MyApp`.
